backend/app/model/digital_object.rb

```diff
@@ -99,14 +102,24 @@ def self.sequel_to_jsonmodel(objs, opts = {})
     jsons
   end
 
+  def self.instance_owners_root_records(id)
+    relationships = self.find_relationship(:instance_do_link).find_by_participant(self[id])
+    resource_ids = Instance.inner_join(:archival_object, archival_object__id: :instance__archival_object_id)
+                     .filter(instance__id: relationships.map {|relationship| relationship[:instance_id]})
+                     .map { |row| row[:root_record_id] }
+    resource_ids
+  end
+
   def delete
+    affected_resource_ids = self.class.instance_owners_root_records(self[:id])
     related_records(:instance_do_link).map {|sub| sub.delete }
+    Resource.update_mtime_for_ids(affected_resource_ids)
     super
   end
 
-
-  repo_unique_constraint(:digital_object_id,
-                         :message => "Must be unique",
-                         :json_property => :digital_object_id)
-
+  def update_from_json(json, opts = {}, apply_nested_records = true)
+    affected_resource_ids = self.class.instance_owners_root_records(self[:id])
+    Resource.update_mtime_for_ids(affected_resource_ids)
+    super
+  end
```

Comment:
It _might_ be preferable (correct?) to call `update_mtime_for_ids` after `super` in these cases? Let the data updates / deletions happen then bump for reindexing as late as possible ... (this also applies [if it does] to digital_object_component.rb).

Reply:
This should be the case now - see next comment